src/include/btmem.h

```diff
@@ -744,12 +744,13 @@ struct __wt_ref {
 #define	ref_ikey	key.ikey
 
 	WT_PAGE_DELETED	*page_del;	/* Deleted on-disk page information */
+	bool has_las_records;
```

Comment:
I would be very hesitant to introduce new states, we can find you a bit.

Off-hand, I'd suggest splitting `WT_REF.state` into two 16-bit fields? I don't think that should slow anything down...